recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -142,6 +142,16 @@ this.PreferenceExperiments = {
       previousPreferenceValue: Preferences.get(preferenceName, undefined),
     };
 
+    if (experiment.previousPreferenceValue) {
+      const previousType = typeof experiment.previousPreferenceValue;
+      const givenType = typeof preferenceValue;
```

Comment:
@Osmose It doesn't seem that I can just give the method a value and have it spit the type of out; it takes a pref name and checks the current value of that pref.

Does that mean that in order to test the new given value, it needs to first set the pref, then check the type, then revert the value if necessary?

I looked through the docs and didn't see anything that would help cut out this weird set-then-reset business, but maybe I missed something. Thoughts?